Lombiq.Hosting.MultiTenancy.Tenants/Handlers/FeaturesEventHandler.cs

```diff
@@ -0,0 +1,75 @@
+using Lombiq.Hosting.MultiTenancy.Tenants.Constants;
+using Lombiq.Hosting.MultiTenancy.Tenants.Models;
+using Microsoft.Extensions.Options;
+using OrchardCore.Environment.Extensions.Features;
+using OrchardCore.Environment.Shell;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Lombiq.Hosting.MultiTenancy.Tenants.Handlers;
+
+public sealed class FeaturesEventHandler : IFeatureEventHandler
+{
+    private readonly IShellFeaturesManager _shellFeaturesManager;
+    private readonly IOptions<AlwaysEnabledFeaturesOptions> _alwaysEnabledFeaturesOptions;
+    private readonly ShellSettings _shellSettings;
+
+    public FeaturesEventHandler(
+        IShellFeaturesManager shellFeaturesManager,
+        IOptions<AlwaysEnabledFeaturesOptions> alwaysEnabledFeaturesOptions,
+        ShellSettings shellSettings)
+    {
+        _shellFeaturesManager = shellFeaturesManager;
+        _alwaysEnabledFeaturesOptions = alwaysEnabledFeaturesOptions;
+        _shellSettings = shellSettings;
+    }
+
+    Task IFeatureEventHandler.InstallingAsync(IFeatureInfo feature) => Task.CompletedTask;
+
+    Task IFeatureEventHandler.InstalledAsync(IFeatureInfo feature) => Task.CompletedTask;
+
+    Task IFeatureEventHandler.EnablingAsync(IFeatureInfo feature) => Task.CompletedTask;
+
+    Task IFeatureEventHandler.EnabledAsync(IFeatureInfo feature) => EnableMediaRelatedFeaturesAsync(feature);
+
+    Task IFeatureEventHandler.DisablingAsync(IFeatureInfo feature) => Task.CompletedTask;
+
+    Task IFeatureEventHandler.DisabledAsync(IFeatureInfo feature) => KeepFeaturesEnabledAsync(feature);
```

Comment:
How do you mean? I have tried to do that by re-routing to the features page if one of the AlwaysEnabled features was getting disabled, but it still went through and the feature got disabled. Is there a way to cancel the current request so that the feature disabling request is actually canceled?

Reply:
Yeah, don't worry I thought DisablingAsync can actually intervene in the disabling of features but it certainly cannot do that. Basically, that event handler is useless:
https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore/Shell/ShellDescriptorFeaturesManager.cs#L75